app_server/app/controllers/UserController.js

```diff
@@ -92,7 +99,21 @@ Class.login = function (request, reply) {
 };
 
 Class.logout = function (request, reply) {
-  reply('Logged out!');
+  request.cookieAuth.clear();
+
+  if (request.auth.credentials) {
+    var credentials = request.auth.credentials;
+    request.server.app.cache.drop(credentials.userId,
+                                  (err) => logger.error(err));
+    Service.updateUserParticulars(credentials.userId, {password: ''})
+    .then((updatedUser) => {
+      if (!updatedUser) {
+        logger.error('Failed to clear password: %j', updatedUser);
+      }
+    });
```

Comment:
Good catch. The password regeneration is giving some serious problems when a user login in different places at the same time while we currently don't have a component to manage all valid passwords a.k.a tokens. I will remove that part for now.